### ISSUE TITLE
[Docs] Improve the demo of autocapitalization

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#3155]
Right now the demo in https://ckeditor.com/docs/ckeditor5/latest/features/text-transformation.html#example-using-transformationsremove-and-extra supports only autocapitalization after `. `. Let's try to mimic the most popular text editors' behavior.

**Comment by godai78** [comment]
Added capitalization after `!` and `?`. We could also really use capitalizing the very first letter in the editor area, but I don't even know where to start with that :)

**Comment by oleq** [comment]
I checked #10387 and it looks fine for what it is. 

To capitalize the first letter in the sentence or the first letter in editor content, we'd need to change the regexp a bit, though. Instead of `/([.?!] )([a-z])$/`, that would be `/(^|[.?!] )([a-z])$/`, which I say works pretty well unless there's an inline widget in the middle of the sentence. Why? The inline widget (e.g. inline image) breaks the paragraph into two text nodes and, if the regex is tested against the latter, typing directly after the image will act like typing at the beginning of the paragraph, so

```
<paragraph>foo <image/>[]</paragraph> // typing "bar"
```

will become

```
<paragraph>foo <image/>Bar[]</paragraph>
```

I'm not sure we want to showcase a buggy behavior like this, although I realize this whole autocapitalization example is a bit naive. The current API (which comes down to just a regex) is simply missing the context to analyze the surroundings of the text node.

TL;DR: I'm merging Bartek's PR. If we want anything more complex or smarter, the text transformation feature needs a better API.

**Comment by godai78** [comment]
I say we keep it like it is then and thanks you :)

**Comment by oleq** [comment]
I created a follow-up about making text transformation smarter #10406.
